File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.py
def solution(N, A, B):
    answer = 0
    num = [A, B]
    num.sort()  # 일단 넣어서 정렬

    small = num[0]
    big = num[1]

    while True:
        if(small % 2 == 1 and big % 2 == 0 and small+1 == big):
            break
        if small % 2 == 0: # 짝수면 
            small = small // 2
        else: # 홀수면
            small = small // 2 + 1
        if big % 2 == 0:
            big = big // 2
        else:
            big = big // 2 + 1
        answer += 1

    return answer + 1

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
import java.util.Arrays;
class Solution
{
    public int solution(int n, int a, int b)
    {
        int answer = 0;
        int[] num = {a, b};
        Arrays.sort(num);

        int small = num[0];
        int big = num[1];

        while(!(small % 2 == 1 && big % 2 == 0 && small+1 == big)){
            if(small % 2 == 0) small = small / 2;
            else small = small / 2 + 1;
            if(big % 2 == 0) big = big / 2;
            else big = big / 2 + 1;
            answer++;
        }
        return answer+1;
    }
}